RTLCSS Fix

SmushyTaco opened the pull request:
This fixes #68364 to allow for the proper use of control directives.

## What?
This PR stops the moving of comments and removal of comments from breaking RTLCSS control directives.

## Why?
RTLCSS control directives use comments and if those comments are removed or moved, this will cause this to not function properly. See #68364 for more information.

## How?
The issue is addressed by removing all comments only after RTLCSS control directives are processed. This is done by processing `cssnano` changes through it's proper webpack plugin.

## Testing Instructions
Build this SCSS code:
```scss
.wp-block-create-block-my-block {
  /*!tedgfdfg*/
  /*!rtl:ignore*/
  text-align: left;
  /*!rtl:remove*/
  background: #CCC;
}
```
RTLCSS control directives will be correctly applied.

### Testing Instructions for Keyboard
Not applicable.

## Screenshots or screencast
Not applicable.

Commit:
RTLCSS Fix

SmushyTaco commented:
@gziolo I'm unsure as to why so many of these tests are failing since the changes I've made are minimal. Maybe you can look into it?

Commit:
Merge branch 'trunk' into rtlcss-fix

shvlv commented:
Hi @SmushyTaco, it seems like you deleted `package-lock.json` or run `npm update`. It introduces unintentional changes. You could try to follow the suggestion https://tkdodo.eu/blog/solving-conflicts-in-package-lock-json, i.e., edit `package.json` and run `npm install` after that. If you need to [rebase](https://developer.wordpress.org/block-editor/contributors/code/git-workflow/#keeping-your-branch-up-to-date) on `trunk`, you can resolve `package.json` conflicts, accept the `theirs` version of `package-lock.json`, and run `npm install --package-lock-only`.

Commit:
Merge branch 'trunk' into rtlcss-fix

Commit:
Fixed package-lock

SmushyTaco commented:
I went ahead and added `.idea` to the `.gitignore` so I don't have to keep manually deleting files that are generated my WebStorm, I hope that isn't a problem.

I also added TODO messages for code that needs updating, simplified some REGEX, removed a redundant initializer, and fixed the syntax of a JSDoc.

Commit:
Merge branch 'trunk' into rtlcss-fix

SmushyTaco commented:
@shvlv @gziolo all tests have passed! This PR is now ready to be merged!

shvlv commented on .gitignore:
```diff
@@ -53,3 +53,6 @@ test/gutenberg-test-themes/twentytwentyone
 test/gutenberg-test-themes/twentytwentythree
 test/gutenberg-test-themes/twentytwentyfour
 packages/react-native-editor/src/setup-local.js
+
+# JetBrains IDE Generated Files (WebStorm)
+.idea
```
You can use the global `excludesfile` git configuration to ignore your platform/IDE-specific files - https://gist.github.com/subfuzion/db7f57fff2fb6998a16c. It might be even more than `.idea`.

SmushyTaco replied:
It's only ever `.idea`, at least from my experience, JetBrains stores all their IDE related stuff in that folder so it's easy to exclude. With that said, I've reverted the `.gitignore`

SmushyTaco committed:
Update .gitignore

Commit:
Merge branch 'trunk' into rtlcss-fix

SmushyTaco commented:
@shvlv the `.gitignore` has been reverted and all tests have passed, this is ready to merge!

SmushyTaco commented:
I don't have permission to add a label to make that test pass, so if you could do it for me that would be greatly appreciated.

Commit:
Version bump

Commit:
Merge branch 'trunk' into rtlcss-fix

Commit:
Merge branch 'trunk' into rtlcss-fix

Commit:
Merge branch 'trunk' into rtlcss-fix

Commit:
Merge branch 'trunk' into rtlcss-fix

SmushyTaco commented:
@shvlv @gziolo can we merge this soon?

Commit:
Merge branch 'trunk' into rtlcss-fix

SmushyTaco committed:
Merge branch 'trunk' into rtlcss-fix

SmushyTaco committed:
RTLCSS Source Maps

SmushyTaco commented:
I did not rebase properly which lead to all those code review requests and I changed my GitHub email which made commits no longer associate to my account, I'll be closing this and opening a new PR to fix this. My apologies.
